e2e/Tests/iothub/service/ConfigurationsClientE2ETests.cs

```diff
@@ -202,7 +203,7 @@ await FluentActions
                     await serviceClient.Configurations.DeleteAsync(configurationId).ConfigureAwait(false);
                 }
                 catch (IotHubServiceException ex)
-                    when (ex.StatusCode is HttpStatusCode.NotFound && ex.ErrorCode is IotHubServiceErrorCode.DeviceNotFound)
+                    when (ex.StatusCode is HttpStatusCode.NotFound && ex.ErrorCode is IotHubServiceErrorCode.ConfigurationNotFound)
```

Comment:
This was causing this catch to let it slip and it logged the general concern in the `Exception` catch.